### PR TITLE
Fix TLS 1.3 Post Handshake Auth

### PR DESCRIPTION
**Pull request by jborean93**
Fixed TLS 1.3 Post Handshake Auth when the server certificate validation was disabled.

Fixes: https://github.com/urllib3/urllib3/issues/3325

### CODE REVIEW

**Comment by jborean93** [comment]
Looks like we can still use a context from pyOpenSSL which doesn't have the `post_handshake_auth` attribute. Re added the `getattr` check from before.

File: src/urllib3/util/ssl_.py [hunk 1]
```diff
@@ -320,13 +320,8 @@ def create_urllib3_context(
...
     # The attribute is None for OpenSSL <= 1.1.0 or does not exist in older
-    # versions of Python. We only enable if certificate verification is enabled to work
-    # around Python issue #37428
-    # See: https://bugs.python.org/issue37428
-    if (
-        cert_reqs == ssl.CERT_REQUIRED
-        and getattr(context, "post_handshake_auth", None) is not None
-    ):
+    # versions of Python.
+    if getattr(context, "post_handshake_auth") is not None:
```
Comment (sethmlarson) [inline src/urllib3/util/ssl_.py:324]: We might still need a default "None" value in getattr here?

File: src/urllib3/util/ssl_.py [hunk 2]
```diff
@@ -320,13 +320,8 @@ def create_urllib3_context(
     # Enable post-handshake authentication for TLS 1.3, see GH #1634. PHA is
     # necessary for conditional client cert authentication with TLS 1.3.
     # The attribute is None for OpenSSL <= 1.1.0 or does not exist in older
```
Comment (sethmlarson) [inline src/urllib3/util/ssl_.py:322]: Can remove comment about older versions of Python.

**Review by sethmlarson — CHANGES_REQUESTED** [review]
Thank you! Few comments:

File: src/urllib3/util/ssl_.py (hunk 1 above)
Comment (jborean93) [reply src/urllib3/util/ssl_.py]: My bad, yes it's still needed.

File: src/urllib3/util/ssl_.py (hunk 2 above)
Comment (jborean93) [reply src/urllib3/util/ssl_.py]: Thanks, I've removed that comment but added in a comment about SSLContext from pyOpenSSL.

**Review by sethmlarson — APPROVED** [review]
LGTM, thank you much!

**Comment by jborean93** [comment]
Thanks for the speedy review and merge!

**Comment by sethmlarson** [comment]
@jborean93 It's easy with a high quality PR, thanks again! We'll try to get a release cut with this fix too. Appreciate you using v2.0 of urllib3 :)

**Pull request merged**